File: MyMiniProject/angular-dotnet-finance/backend/Controllers/AuthController.cs
using System.Security.Claims;
using FinanceApi.Data;
using FinanceApi.Models;
using FinanceApi.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace FinanceApi.Controllers;

[ApiController]
[Route("api/[controller]")]
public class AuthController(AppDbContext db, JwtService jwt) : ControllerBase
{
    [HttpPost("login")]
    [AllowAnonymous]
    public async Task<ActionResult<LoginResponse>> Login([FromBody] LoginRequest req)
    {
        if (string.IsNullOrWhiteSpace(req.Username) || string.IsNullOrWhiteSpace(req.Password))
            return BadRequest(new { message = "กรุณากรอกชื่อผู้ใช้และรหัสผ่าน" });

        var user = await db.Users.FirstOrDefaultAsync(u => u.Username == req.Username);
        if (user is null || !user.IsActive || !BCrypt.Net.BCrypt.Verify(req.Password, user.PasswordHash))
            return Unauthorized(new { message = "ชื่อผู้ใช้หรือรหัสผ่านไม่ถูกต้อง" });

        var token = jwt.CreateToken(user);
        return Ok(new LoginResponse(token, ToDto(user)));
    }

    [HttpGet("me")]
    [Authorize]
    public async Task<ActionResult<UserDto>> Me()
    {
        var id = int.Parse(User.FindFirstValue(ClaimTypes.NameIdentifier)!);
        var user = await db.Users.FindAsync(id);
        if (user is null) return Unauthorized();
        return Ok(ToDto(user));
    }

    public static UserDto ToDto(User u) =>
        new(u.Id, u.Username, u.FullName, u.Email, u.Role, u.IsActive, u.CreatedAt);
}

File: MyMiniProject/laravel-finance/app/Http/Controllers/Api/AuthController.php
<?php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\User;
use App\Services\JwtService;
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Hash;

class AuthController extends Controller
{
    public function __construct(private JwtService $jwt) {}

    public function login(Request $request)
    {
        $username = trim((string) $request->input('username', ''));
        $password = (string) $request->input('password', '');

        if ($username === '' || $password === '') {
            return response()->json(['message' => 'กรุณากรอกชื่อผู้ใช้และรหัสผ่าน'], 400);
        }

        $user = User::where('username', $username)->first();
        if (! $user || ! $user->is_active || ! Hash::check($password, $user->password)) {
            return response()->json(['message' => 'ชื่อผู้ใช้หรือรหัสผ่านไม่ถูกต้อง'], 401);
        }

        return response()->json([
            'token' => $this->jwt->createToken($user),
            'user' => $user->toApiArray(),
        ]);
    }

    public function me(Request $request)
    {
        /** @var User $user */
        $user = $request->attributes->get('auth_user');

        return response()->json($user->toApiArray());
    }
}
